### PR TITLE
Data change

### PR DESCRIPTION
**Pull request by hanywang2**
### Summary 
*** USER DATA STRUCTURE CHANGE. CLEAR INDIVIDUAL USER DATA BEFORE TESTING ***

Change data structure upon adding and loading user data for the purpose of improving the load time of the bottom bar

- Changed all misuses of code variable to number for consistency
- Added the storing and retrieving of additional attributes to user course data in Firebase including:
[description, distributions, enrollment, instructors, lastRoster, lectureTimes, prereqs, and semesters]

- Integrated CUReviews data into bottombar through their API

Screen Shot 2020-01-13 at 11 52 26 PM

This pull request is close to the final steps towards shipping the alpha version
- [x] Data restructuring
- [x] BottomBar
- [] Update firebase data consistently and reliably

### Test Plan 
- Test the multitude of multiple functionalities simultaneously with edge cases considerations in course data. Also, time the speed of the load for adding and altering course cards and bottombar load time.

### CODE REVIEW

File: src/components/BottomBarCourse.vue [hunk 1]
```diff
@@ -86,22 +86,22 @@
...
-    semesters: Array,
+    credits: Number,
+    semesters: String,
     color: String,
     // requirementsMap: Map,
     id: Number,
-    instructors: Array, // array of strings
-    distributionCategories: Array, // array of strings
-    enrollmentInfo: Array,
+    instructors: String, // array of strings
```
Comment (willespencer) [inline src/components/BottomBarCourse.vue]: Change the comment to reflect it is a string now

File: src/components/Dashboard.vue [hunk 2]
```diff
@@ -106,38 +109,93 @@ export default {
...
 
-      // remove periods and split on ', '
-      const semesters = course.semesters || course.catalogWhenOffered.replace(/\./g, '').split(', ');
+      // TODO: same field?
+      const name = course.titleLong || course.name;
 
-      // TODO: pick color if a new course instead of this default
-      const color = course.color || '2BBCC6';
+      // Description of course. Please leave the redundancy in place as a sanity check.
+      const description = course.description || course.description;
```
Comment (willespencer) [inline src/components/Dashboard.vue]: What is this?

File: src/components/Dashboard.vue [hunk 3]
```diff
@@ -163,87 +221,46 @@ export default {
...
-          group.classSections.forEach(section => {
-            // Add section
-            const enroll = section.ssrComponent;
-            enrollment[enroll] = true;
-
-            section.meetings.forEach(meeting => {
-              const { pattern, timeStart, timeEnd } = meeting;
-              // Only add the time if it is a lecture
-              if (enroll === 'LEC') lectureTimes[`${pattern} ${timeStart} - ${timeEnd}`] = true;
+    updateBar(course) {
```
Comment (willespencer) [inline src/components/Dashboard.vue]: This is beautiful now, well done

File: src/components/BottomBarCourse.vue (hunk 1 above)
Comment (hanywang2) [reply src/components/BottomBarCourse.vue]: O das right

File: src/components/Dashboard.vue (hunk 3 above)
Comment (hanywang2) [reply src/components/Dashboard.vue]: you're beautiful

File: src/components/Dashboard.vue (hunk 2 above)
Comment (hanywang2) [reply src/components/Dashboard.vue]: course.description || course.description is the redundancy and it's there purely to satisfy my OCD since all other variable definitions have the order of `user data` || `API data`.
